app/uk/gov/hmrc/economiccrimelevyregistration/services/EclRegistrationService.scala

```diff
@@ -48,4 +50,79 @@ class EclRegistrationService @Inject() (
 
   def upsertRegistration(registration: Registration)(implicit hc: HeaderCarrier): Future[Registration] =
     eclRegistrationConnector.upsertRegistration(registration)
+
+  def getSubscription(eclRegistration: String)(implicit hc: HeaderCarrier): Future[GetSubscriptionResponse] =
+    eclRegistrationConnector.getSubscription(eclRegistration)
+
+  def transformToRegistration(
+    registration: Registration,
+    getSubscriptionResponse: GetSubscriptionResponse
+  ): Registration = {
+    val primaryContact      = getSubscriptionResponse.primaryContactDetails
+    val secondaryContact    = getSubscriptionResponse.secondaryContactDetails
+    val subscriptionAddress = getSubscriptionResponse.correspondenceAddressDetails
+
+    val firstContactDetails: ContactDetails = ContactDetails(
+      Some(primaryContact.name),
+      Some(primaryContact.positionInCompany),
+      Some(primaryContact.emailAddress),
+      Some(primaryContact.telephone)
+    )
+    val secondContactDetails                = secondaryContact match {
+      case Some(value) =>
+        ContactDetails(
+          Some(value.name),
+          Some(value.positionInCompany),
+          Some(value.emailAddress),
+          Some(value.telephone)
+        )
+      case _           => ContactDetails.empty
+    }
+    val secondContactPresent                = if (secondaryContact.isEmpty) {
+      Some(false)
+    } else {
+      Some(true)
+    }
+
+    val contacts: Contacts = Contacts(firstContactDetails, secondContactPresent, secondContactDetails)
+
+    val businessSector: BusinessSector =
+      BusinessSector.transformFromSubscriptionResponse(getSubscriptionResponse.additionalDetails.businessSector)
+    val address: EclAddress            = EclAddress(
+      None,
+      Some(subscriptionAddress.addressLine1),
+      subscriptionAddress.addressLine2,
+      subscriptionAddress.addressLine3,
+      subscriptionAddress.addressLine4,
+      None,
+      subscriptionAddress.postCode,
+      None,
+      subscriptionAddress.countryCode.getOrElse(
+        throw new IllegalStateException("Country code not found in GetSubscriptionResponse")
+      )
+    )
+    registration.copy(
+      contacts = contacts,
+      businessSector = Some(businessSector),
+      contactAddress = Some(address),
+      partnershipName = getSubscriptionResponse.legalEntityDetails.organisationName,
+      amlSupervisor = Some(getAmlSupervisor(getSubscriptionResponse.additionalDetails.amlSupervisor))
+    )
+  }
+
+  private def getAmlSupervisor(amlSupervisor: String): AmlSupervisor = {
+    val amlProfessionalBodySupervisors: Option[String] =
+      appConfig.amlProfessionalBodySupervisors.find(p =>
+        p.toLowerCase() == amlSupervisor.replace(" ", "").toLowerCase()
+      )
+
+    if (amlProfessionalBodySupervisors.isEmpty) {
```

Comment:
if we need to strip spaces from amlSupervisor twice and lowercase, we should save that in a val

Reply:
Done